src/Controller/CRUDController.php

```diff
@@ -996,22 +1108,14 @@ protected function configure(): void
         }
 
         try {
-            $this->admin = $this->container->get('sonata.admin.pool')->getAdminByAdminCode($adminCode);
+            $this->admin = $this->pool->getAdminByAdminCode($adminCode);
         } catch (\InvalidArgumentException $e) {
             throw new \RuntimeException(sprintf(
                 'Unable to find the admin class related to the current controller (%s)',
                 static::class
             ));
         }
 
-        $this->templateRegistry = $this->container->get(sprintf('%s.template_registry', $this->admin->getCode()));
```

Comment:
this template registry has to be the template registry for this admin.

Reply:
I don't understand what are you waiting for

Reply:
I think it would be better to use: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Admin/AbstractAdmin.php#L3214 directly from the admin

Reply:
Sorry i don't understand. You point out a removed line. For me, none AbstractAdmin is available in the CRUDController

Reply:
https://github.com/sonata-project/SonataAdminBundle/blob/master/src/Admin/AbstractAdmin.php#L2398

It is not removed

Reply:
I'm talking about the line above. (https://github.com/sonata-project/SonataAdminBundle/pull/6093/files/6e6368b7dc8dfa1f650ad98818a0361c4c313f42#diff-a8a33ea35d3afb5f4ce4a69f18516b03L1007)

Reply:
In CRUDController, this line is remove. I don't use templateRegistry anymore here : https://github.com/sonata-project/SonataAdminBundle/pull/6093/files/6e6368b7dc8dfa1f650ad98818a0361c4c313f42#diff-a8a33ea35d3afb5f4ce4a69f18516b03L1007

Reply:
Then you should remove it from construct right?

Reply:
No... so the thing is: you are injecting a global template registry but each admin has his own with possibly different templates for each admin. Thats what we should restore, you should recover this lines (with my comment applied) and remove the construct template registry

Reply:
AdminInterface doesn't implement getTemplateRegistry. Should i add this in the interface ?

Reply:
Well, no it is not that easy, because the method is protected. this was implemented here: https://github.com/sonata-project/SonataAdminBundle/pull/4980

And I think it was before the public / private container thing. We need to think a better solution here.

@sonata-project/contributors help! Do we make the getTemplateRegistry public here? I don't see a lot of better alternatives. I think using a service locator won't work here, because we don't know the correct templateregistry when doing the dependency injection.

Reply:
We need to use this somehow: https://symfony.com/doc/current/service_container/service_subscribers_locators.html

I will try to investigate a bit tomorrow on how we could do it, probably will be opening a separate PR to prepare the ground for this one.

Reply:
Normally, we extend from AbstractController so we already had subscribed services. I really don't understand why it is not works.